config/webpack/theme-linter-plugin/ThemeLinterPlugin.js

```diff
@@ -0,0 +1,81 @@
+const postcss = require('postcss');
+const postcssValueParser = require('postcss-value-parser');
+const getThemeConfig = require('../postcss/_getThemeConfig');
+
+/**
+ * This plugin provides a post css loader plugin that tracks themeable variables and a webpack plugin that aggregates those
+ * results as warnings by the webpack compilation
+ */
+module.exports = class ThemeLinterPlugin {
+  constructor(config) {
+    const themeConfig = config || getThemeConfig();
+
+    this.variableInformation = {
+      themeableVariables: new Set(),
+      themeVariableTracker: {},
+    };
+
+    // Set up a tracker for each theme in the config
+    const themes = [themeConfig.theme, ...(themeConfig.scoped || [])].filter((theme) => theme !== undefined);
+    themes.forEach((theme) => {
+      this.variableInformation.themeVariableTracker[theme] = {
+        populatedVariables: new Set(),
+        duplicateVariableTracker: {},
+      };
+    });
+  }
+
+  loaderPlugin() {
+    return postcss.plugin('terra-theme-linter-plugin', () => (
+      (root) => {
+        // Walk the declarations and flag all values that are var's
+        root.walkDecls(decl => {
+          postcssValueParser(decl.value).walk(node => {
+            if (node.type === 'function' && node.value === 'var') {
+              this.variableInformation.themeableVariables.add(node.nodes[0].value);
+            }
+          });
+        });
+
+        // For each theme, walk each rule that is declared as a class for that theme and then walk declarations
+        // for that rule and mark the variables as tracked (or duplicate if they've already been added)
+        Object.entries(this.variableInformation.themeVariableTracker).forEach(([theme, tracker]) => {
+          root.walkRules(RegExp(`.${theme}`), (node) => {
+            node.walkDecls(decl => {
+              const updatedTracker = tracker;
+              updatedTracker.populatedVariables.add(decl.prop);
+              updatedTracker.duplicateVariableTracker[decl.prop] = tracker.duplicateVariableTracker[decl.prop] || new Set();
+              updatedTracker.duplicateVariableTracker[decl.prop].add(decl.value);
+            });
+          });
+        });
+      }
+    ));
+  }
+
+  apply(compiler) {
+    compiler.hooks.emit.tapPromise('TerraThemeLinterPlugin', (compilation) => {
+      Object.entries(this.variableInformation.themeVariableTracker).forEach(([theme, tracker]) => {
+        // Mark variables that are in the list of themeable variables but weren't tracked as a missing warning
+        const missingThemeVariables = Array.from(new Set([...this.variableInformation.themeableVariables].filter(x => !tracker.populatedVariables.has(x)))).sort();
```

Comment:
this is a lot of logic on one line - what are your thoughts on splitting the inner set into a variable? 
```suggestion
        const = missingThemeVariablesSet = new Set([...this.variableInformation.themeableVariables].filter(x => !tracker.populatedVariables.has(x)));
        const missingThemeVariablesArr = Array.from(missingThemeVariablesSet).sort();
```